tests/unit/test_collector.py

```diff
@@ -41,3 +168,71 @@ def test_01_mega_raid_collector_installed_and_okay(self):
         self.assertEqual(len(list(payloads)), len(available_metrics))
         for payload in payloads:
             self.assertIn(payload.name, available_metrics)
+
+    def test_10_ipmi_dcmi_collector_not_installed(self):
+        """Test ipmi dcmi collector when ipmi-dcmi is not installed."""
+        ipmi_dcmi_collector = IpmiDcmiCollector()
+        payloads = ipmi_dcmi_collector.collect()
```

Comment:
The command should actually be mocked. Because without it, the `collect()` will try to run the command under user's local environment (credit: @jneo8 ) Same goes to other tests.

Reply:
The external input should always be mocked to avoid the unexpected side effect.